### PR TITLE
cleanup cmake files

### PR DESCRIPTION
**Pull request by ClausKlein**
cmaklint this files too
detect standalone build and build test and examples too

fix too https://github.com/kokkos/mdspan/issues/37

### CODE REVIEW

File: cmake/googletest/CMakeLists.txt.in [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # from https://github.com/google/googletest/blob/master/googletest/README.md
-cmake_minimum_required(VERSION 2.8.2)
+cmake_minimum_required(VERSION 3.8)
```
Comment (dalg24) [inline cmake/googletest/CMakeLists.txt.in]: Why did you change?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
+cmake_minimum_required(VERSION 3.12...3.19)
```
Comment (dalg24) [inline CMakeLists.txt:1]: Why would we specify a maximum version?

File: CMakeLists.txt [hunk 3]
```diff
@@ -11,8 +11,18 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
...
-option(MDSPAN_ENABLE_TESTS "Enable tests." Off)
-option(MDSPAN_ENABLE_EXAMPLES "Build examples." Off)
+
+# Determine if MDSpan is built as a subproject (using add_subdirectory)
+# or if it is the master project.
+set(MASTER_PROJECT OFF)
+if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  set(MASTER_PROJECT ON)
+  message(STATUS "CMake version: ${CMAKE_VERSION}")
+endif()
```
Comment (dalg24) [inline CMakeLists.txt]: Why would we change the default if it is build as a standalone.  What was wrong with `OFF`?

File: examples/CMakeLists.txt [hunk 4]
```diff
@@ -1,8 +1,15 @@
+cmake_minimum_required(VERSION 3.12...3.19)
```
Comment (dalg24) [inline examples/CMakeLists.txt:1]: Same comment here.  I feel like specifying a maximum version is not a good idea.

File: CMakeLists.txt [hunk 5]
```diff
@@ -114,38 +124,38 @@ target_include_directories(mdspan INTERFACE
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include>
 )
+target_compile_features(mdspan INTERFACE cxx_std_${CMAKE_CXX_STANDARD})
 
 ################################################################################
 
-install(TARGETS mdspan EXPORT mdspanTargets
+install(TARGETS mdspan EXPORT MDSpanTargets
```
Comment (dalg24) [inline CMakeLists.txt]: No I think `mdspan` was right if the package name is "mdspan".
The intent is user will do `find_package(mdspan)` and not `find_package(MDSpan)`

Comment (dalg24) [inline CMakeLists.txt]: No I think `mdspan` was right if the package name is "mdspan".
The intent is user will do `find_package(mdspan)` and not `find_package(MDSpan)`

**Review by dalg24 — CHANGES_REQUESTED** [review]
Looks good other than the maximum required CMake version and the change of the package name

File: cmake/googletest/CMakeLists.txt.in (hunk 1 above)
Comment (ClausKlein) [reply cmake/googletest/CMakeLists.txt.in]: because support for cmake versions 2.8 is deprecated and `ExternalProject_Add` is available starting with cmake version 3.0. 
see https://cmake.org/cmake/help/latest/release/3.0.html?highlight=externalproject

File: CMakeLists.txt (hunk 2 above)
Comment (ClausKlein) [reply CMakeLists.txt:1]: There is not a maximum version!
This mean that the newest cmake policies between this ranch is used.
Depending of the cmake version used for build.

File: CMakeLists.txt (hunk 3 above)
Comment (ClausKlein) [reply CMakeLists.txt]: This is only to prevent to build tests and examples if using as a subproject with FetchContents or as git submodule from an other cmake project. And IMHO the standalone project should build the test and its examples.

File: CMakeLists.txt (hunk 5 above)
Comment (ClausKlein) [reply CMakeLists.txt]: The project name is `MDSpan`, the library name is `mdspan` and the namespace is `std`.
Normale the project name and the namespace is the same. But IMHO, it must not.
see https://cmake.org/cmake/help/latest/command/find_package.html?highlight=find_package#id3
I change this.

File: CMakeLists.txt (hunk 3 above)
Comment (dalg24) [reply CMakeLists.txt]: > This is only to prevent to build tests and examples if using as a subproject with FetchContents or as git submodule from an other cmake project.

This is how it behave previously.  Turned off by default.

> And IMHO the standalone project should build the test and its examples.

I tend to prefer the current behavior.

**Review by dalg24 — APPROVED** [review]
Fine now even though I disagree with the change in the option the default values

File: CMakeLists.txt (hunk 3 above)
Comment (ClausKlein) [reply CMakeLists.txt]: OK, done

File: CMakeLists.txt (hunk 5 above)
Comment (ClausKlein) [reply CMakeLists.txt]: Ok, done
